internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -696,17 +697,31 @@ func (r *NumaflowControllerRolloutReconciler) processNumaflowControllerStatus(co
 }
 
 func (r *NumaflowControllerRolloutReconciler) markRolloutPaused(ctx context.Context, rollout client.Object, paused bool) error {
+
 	controllerRollout := rollout.(*apiv1.NumaflowControllerRollout)
 
 	if paused {
+		if controllerRollout.Status.PauseRequestStatus.LastPauseBeginTime == metav1.NewTime(initTime) || !controllerRollout.Status.PauseRequestStatus.LastPauseBeginTime.After(controllerRollout.Status.PauseRequestStatus.LastPauseEndTime.Time) {
```

Comment:
please add comments

Reply:
Used same comments as PipelineRollout since they're the same logic.